pkg/tfbridge/diff_test.go

```diff
@@ -1474,3 +1496,191 @@ func TestCollectionsWithMultipleItems(t *testing.T) {
 		}
 	})
 }
+
+func TestSetNestedAddReplace(t *testing.T) {
+	diffTest(t,
+		map[string]*schema.Schema{
+			"prop": {
+				Type: schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"nest": {Type: schema.TypeString, Required: true},
+					},
+				},
+				ForceNew: true,
+			},
+			"outp": {Type: schema.TypeString, Computed: true},
+		},
+		map[string]*SchemaInfo{},
+		map[string]interface{}{
+			"prop": []interface{}{map[string]interface{}{"nest": "baz"}},
+		},
+		map[string]interface{}{
+			"prop": nil,
+			"outp": "bar",
+		},
+		map[string]DiffKind{
+			"prop":         AR,
+			"prop[0].nest": A,
+		})
+}
+
+func TestListNestedAddReplace(t *testing.T) {
+	diffTest(t,
+		// tfSchema
+		map[string]*schema.Schema{
+			"prop": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"nest": {Type: schema.TypeString, Required: true},
+					},
+				},
+				ForceNew: true,
+			},
+			"outp": {Type: schema.TypeString, Computed: true},
+		},
+		// info
+		map[string]*SchemaInfo{},
+		// inputs
+		map[string]interface{}{
+			"prop": []interface{}{map[string]interface{}{"nest": "foo"}},
+		},
+		// state
+		map[string]interface{}{
+			"prop": nil,
+			"outp": "bar",
+		},
+		// expected
+		map[string]DiffKind{
+			"prop":         AR,
```

Comment:
Out of curiosity, have you checked to see how this renders on the CLI side?

Reply:
Just tried it out. Looks fine to me 🤷 

<img width="1351" alt="Screen Shot 2021-01-22 at 8 46 53 PM" src="https://user-images.githubusercontent.com/24326455/105567943-0e338d80-5cf3-11eb-9692-87b877d69366.png">

Reply:
Here's another... this one falls in the `TestCollectionsWithMultipleItems.RemoveFirst` for a set.

<img width="891" alt="Screen Shot 2021-01-25 at 9 18 07 AM" src="https://user-images.githubusercontent.com/24326455/105733720-135d2c00-5eef-11eb-8b82-7f17b255bdf3.png">

Reply:
Awesome. I'll admit that it feels a little weird that we don't see simply `"prop[0].nest": AR,`, but I think that's okay given the rendering.

Reply:
Fixed this in the latest commit.